lib/os/rb.c

```diff
@@ -132,7 +133,7 @@ static void rotate(struct rbnode **stack, int stacksz)
 	struct rbnode *child = stack[stacksz - 1];
 	int side = get_side(parent, child);
 	struct rbnode *a = get_child(child, side);
-	struct rbnode *b = get_child(child, !side);
+	struct rbnode *b = get_child(child, side == 0 ? 1 : 0);
```

Comment:
Should we perhaps have get_side return a bool?

Reply:
Or write an other_side() predicate for usage like this.  Though this is clear enough.